lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1622,6 +1624,93 @@ def get_configured_disk_size(self, expected_disk_spec):
         self.module.fail_json(
             msg="No size, size_kb, size_mb, size_gb or size_tb attribute found into disk configuration")
 
+    def vmdk_disk_path_split(self, vmdk_path):
```

Comment:
To add this logic to vmware_guest_disk, this method will move into module_utils/vmware.py.